Add guide on things to do when creating new component (#2828)
BBC-archive/psammead: Things-to-do-when-creating-a-new-component.md

  - How will this component behave if JavaScript is disabled? Core content needs to be accessible with JavaScript disabled - ensure that this component allows for this.
  - As a group, break down the work required into small subtasks, and add these to a checklist in the GitHub issue\*. If a subtask requires more effort, members of the swarm can either all work together, or pair on the task. Smaller tasks can be picked up individually. Each of these subtasks can go into code review on their own, rather than waiting for all of the work to be done. You may think of new tasks once you’ve started developing, so feel free to keep adding to this list once you’ve started.
  - Make sure that accessibility requirements are considered throughout this process, not just at the end during the accessibility swarm.
  - Some things to consider that are often missed until code review/testing are high contrast mode in firefox, support for right-to-left languages, and IE support.

> Do we want to do this, or create a new issue for each subtask?

Reviewer: ```suggestion

```

| Write e2e tests (Think about e2es)                                            | end? (start)                                             |
| Think about AMP                                                               | start                                                    |
| Think about non-JS                                                            | start                                                    |
| Involve testers in early discussions of component                             | start                                                    |
| Consider accessibility early on                                               | start                                                    |
| Requirements                                                                  | start                                                    |
| Acceptance criteria                                                           | start                                                    |
| Be clear about where requirements and ACs live - Jira/Zeplin/Docs/Dropbox etc | all the way through                                      |

> | Translations requirements short/long term                                     |                                                          |

Reviewer: ```suggestion
| Translations requirements short/long term                                     |    start/end                                                      |
```

| Split high level requirements into subtasks                                   | During dev swarm                                         |
| Confirm that the designs are up to date                                       | start                                                    |
| Does component require event tracking?                                        | start (once event tracking is complete)                  |
| Decide where this should be developed in pair/individually                    | start                                                    |
| Discuss unit test implementation within refinement                            | Refinement                                               |
| Do we extend the swarming approach beyond devs (include design/test too)?     | start                                                    |

> | Dont forget IE and rtl and high contrast mode FF                              |                                                          |

Reviewer: ```suggestion
| Don't forget about IE, `rtl` and High Contrast mode in FF                              |  all the way through |
```
